Update Docker setup for testing and local development

chubberlisk opened the pull request:
## Describe this PR

### *What is the problem we're trying to solve*

Our Docker setup for testing and local development is a bit out of sync and some naming doesn't quite align.

### *What changes have we introduced*

- Rename the database name for dev and test to `socialcare` instead of `testsocialcare` as this name is used for both dev and test, not just for testing due the setup using connection string etc.
- Rename Make command for migrating the test database from `make migrate-local-test-database` to `make migrate-test-db` to shorten and have aligned with other Make commands for the test database.
- Update docker-compose to expose dev database at port `7654` to allow us to easily access it from a Postgres client.
- Update the `make serve` command to run migrations for dev database.

### Guidance to review

Would recommend reviewing commit by commit.

Commit:
Rename database name for dev and test to "socialcare"

Instead of "testsocialcare" as this is the name used for both dev
and test, not just for testing.

Commit:
Rename Make command for migrating the test database

This shortens and has the naming aligned with other Make commands
for the test database

Commit:
Update docker-compose to expose dev database at port 7654

This will allows us to easily access the dev database from a Postgres
client.

Commit:
Update Make serve command to run migrations for dev database

Renny4Real reviewed (approved):
Looks good 🎉 

I'm a bit unsure of removing `test` from the name of the DB. 😅 

Wondering if having `test`, could act as an identifier that this is for local testing, in the same way in AWS, we have the DB named after the environment.

Renny4Real commented:
@chubberlisk Tried to run the make command with no containers, and the migration command errored.
Looks like we weren't making sure the `dev database` was up before running the migration. 
Is it alright if I add a commit to this PR to fix this?

Renny4Real committed:
Fix migration error when dev database container does not exist

If the docker container for the dev database did not already exist,
when running make serve, the migration errors.

This rearranges the sequence of events in the make serve command so that the database container is up first before running the migration command.

chubberlisk commented:
> @chubberlisk Tried to run the make command with no containers, and the migration command errored.
> Looks like we weren't making sure the `dev database` was up before running the migration.
> Is it alright if I add a commit to this PR to fix this?

Yeah, feel free to - thanks!

Renny4Real commented:
@chubberlisk I've pushed up the commit but wasn't sure if you wanted to check it first before merging? 🙂

chubberlisk commented:
> @chubberlisk I've pushed up the commit but wasn't sure if you wanted to check it first before merging? 🙂

Yup, was going to do this morning! Wanted John's visits PR to get in before merging this in in case it caused problems for him.

Commit:
Update formatting by running linter
